src/silx/gui/hdf5/Hdf5TreeModel.py

```diff
@@ -573,9 +576,12 @@ def __areH5pyObjectEqual(obj1, obj2):
         if not isinstance(obj2, type(obj1)):
             return False
         def key(item):
-            if item.file is None:
-                return item.name
-            return item.file.filename, item.file.mode, item.name
+            info = [item.name]
+            if item.file is not None:
+                info += [item.file.filename, item.file.mode]
+            if isinstance(item, DatasetSlice):
+                info.append(item.indices)
```

Comment:
Same, it'd be best to avoid specific code here.
I wanted to rework the way to check equality, but that's complex changes